### PR TITLE
new dockerify script which builds a cloudify manager in a docker centos (+systemd+ssh) container

### PR DESCRIPTION
**Pull request by funkyHat**
A little rough around the edges, but not sure how polished this needs to be. Feedback welcome :)

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# dockerify.py
+
+In order to quickly test changes to manager components, you can use the `dockerify.py` script to spin up a manager as a docker container.
+
+
+## Prerequisites:
```
Comment (idanmo) [inline scripts/README.md]: Cloudify CLI (`cfy`) in path?

File: scripts/dockerify.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+from subprocess import check_call, check_output, CalledProcessError
+from time import sleep
+
+import yaml
+
+
+EXPOSE = [22, 80, 443, 5671]
+
+# 22 isn't needed here as the user can access SSH via the internal address
+PUBLISH = ['80:80', '443:443', '5671:5671']
```
Comment (dankilman) [inline scripts/dockerify.py]: i don't think any port should be published. exposing them should suffice

File: scripts/dockerify.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        description="Deploy a simple manager blueprint "
+                    "into a local docker container",
+        )
+    parser.add_argument('path', nargs='?', default='.')
+    parser.add_argument(
+        '--ssh-key',
+        default=os.path.expanduser('~/.ssh/id_rsa'),
+        )
+    parser.add_argument(
+        '--docker_tag',
```
Comment (dankilman) [inline scripts/dockerify.py]: arguments usually use `-` and not `_`

File: scripts/dockerify.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        default='cloudify/centos-manager:7',
+        )
+    parser.add_argument(
+        '--docker_context',
+        default=os.path.join(os.path.dirname(__file__), 'dockerify'),
+        )
+    parser.add_argument(
```
Comment (dankilman) [inline scripts/dockerify.py]: you don't seem to be using this flag

File: scripts/dockerify.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+def ssh_swap(id, ip, keyname):
+    """
+    You get the host key in, the client key out
+    in out in out shake it all about
+    """
+    server_pubkey_file = '/etc/ssh/ssh_host_rsa_key.pub'
+    # once this file exists the SSHD in the container is sufficiently ready
+    server_pubkey = _wait_for_file(id, server_pubkey_file)
+    with open(os.path.expanduser('~/.ssh/known_hosts'), 'a+') as f:
```
Comment (dankilman) [inline scripts/dockerify.py]: what are we actually automatically appending this file? 
at most, i would use `ssh-keygen -R` to remove a previous entry if one already exists. All it saves is the simple step of approving a new host

File: scripts/dockerify.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    docker.cp([
+            keyname + '.pub',
+            '{id}:/root/.ssh/authorized_keys'.format(id=id)
+            ])
+
+
+def install(path, id, ip, key_filename):
+
+    # Write the inputs file
+    with open('docker-bootstrap-inputs.yaml', 'w') as f:
```
Comment (dankilman) [inline scripts/dockerify.py]: use tempfile to create a temporary inputs file. currently it is written to the current directory.

File: scripts/dockerify.py (hunk 2 above)
Comment (funkyHat) [inline scripts/dockerify.py]: Doesn't this (at least 5671) mean that the manager can't usefully deploy an agent to an external platform?

File: scripts/dockerify.py (hunk 6 above)
Comment (dankilman) [inline scripts/dockerify.py]: there are cases where you may want to override some inputs so you should allow a user to supply path to inputs and override these four properties if one is provided

File: scripts/dockerify.py (hunk 2 above)
Comment (funkyHat) [inline scripts/dockerify.py]: (and I should probably have 5672 in there as well in that case...)

Comment (dankilman) [inline scripts/dockerify.py]: that is not the intended use case. the intended use case is to start a manager container on the developers laptop for fast iterations

File: scripts/dockerify.py (hunk 6 above)
Comment (funkyHat) [inline scripts/dockerify.py]: mkdtemp added & it will now pass through any `-i|--inputs` args to `cfy`

**Comment by dankilman** [comment]
migrated to https://github.com/dankilman/docl as a standalone tool

**Pull request closed without merging**